mods/default/functions.lua

```diff
@@ -348,9 +348,7 @@ minetest.register_abm({
 	label = "Grass spread",
 	nodenames = {"default:dirt"},
 	neighbors = {
-		"default:dirt_with_grass",
-		"default:dirt_with_dry_grass",
-		"default:dirt_with_snow",
+		"air",
```

Comment:
This is the big optimisation mentioned in tenplus1's PR:
"Removing 'dirt_with_grass' etc. from 'neighbors' stops the ABM action
running everywhere and constantly, on the dirt nodes immediately below
the surface nodes. Now the action only runs in the rare case of a dirt
node with neighbouring air, grass decorations or snow."
If 'group:spreading_dirt_type' was included in neighbours, normal unaltered biomes would trigger a large number of ABM actions on the dirt nodes just below the surface, each action being a light check which would 'return' due to darkness.
So now the slow Lua ABM action only runs when spreading is needed.
'air' is needed in neighbours to trigger for exposed dirt.